Opt fixes and specifics strategy simplification

d-michail opened the pull request:
Main changes
  - Simplified the specifics strategy interface and implementations
  - Fixed a bug when using the FastUtil map classes (we were using reference equality instead of .equals())
  - Added a graph implementation specialized for Integer vertices which uses the more specialized integer keyed map implementation of fastutil.
  - Moved all classes using FastUtil into a separate `.fastutil` package.

@jsichi I am having some trouble naming the graph classes. I used the pattern `FastutilMap - IntegerAny - Graph` where the Integer is the vertices type and the Any is the edges type. Does this make any sense? What about the first part? Should I shorten it, remove it, etc?

----

- [x] I read and understood <https://github.com/jgrapht/jgrapht/wiki/Become-a-Contributor>
- [x] I read and understood <https://github.com/jgrapht/jgrapht/wiki/How-to-make-your-first-%28code%29-contribution>
- [x] I added [unit tests](https://github.com/jgrapht/jgrapht/wiki/Unit-testing)
- [x] I added [documentation](https://github.com/jgrapht/jgrapht/wiki/How-to-write-documentation)
- [x] I followed the [Coding and Style Conventions](https://github.com/jgrapht/jgrapht/wiki/Coding-and-Style-Conventions)
- [x] I **have not** modified `HISTORY.md` or `CONTRIBUTORS.md`
- [x] I ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)

jsichi commented:
How about this convention:  

* FastutilMapIntVertexGraph (implies edge type is unrestricted)
* FastutilMapIntVertexLongEdgeGraph (edge type is specialized to long)

(Let's follow the fastutil typename convention Int, Long, etc)

For GraphSpecificsStrategy, acronymization to GSS is probably called for.

d-michail commented:
I renamed the classes, per your suggestion. I also moved a few tests from the new isomorphism package to the slow tests category, because the test phase was getting slow again.

jsichi commented on jgrapht-opt/src/main/java/org/jgrapht/opt/graph/fastutil/FastutilMapGraph.java:
```diff
@@ -7,11 +7,24 @@
+ * <pre>
+ * Graph&lt;Integer,
+ *     DefaultWeightedEdge&gt; g = new FastutilMapGraph&lt;&gt;(
+ *         SupplierUtil.createIntegerSupplier(), SupplierUtil.createDefaultWeightedEdgeSupplier(),
```
Better to use createStringSupplier for this example since for int we steer them to use FastutilMapIntVertexGraph below.

jsichi commented:
Thanks!
